MultiServer.py

```diff
@@ -643,8 +643,12 @@ async def on_client_connected(ctx: Context, client: Client):
         'hint_cost': ctx.hint_cost,
         'location_check_points': ctx.location_check_points,
         'datapackage_version': network_data_package["version"],
-        'datapackage_versions': {game: game_data["version"] for game, game_data
-                                 in network_data_package["games"].items()},
+        'datapackage_versions': {
+            game: game_data["version"] for game, game_data in network_data_package["games"].items()
+        },
+        'datapackage_checksums': {
+            game: game_data["checksum"] for game, game_data in network_data_package["games"].items()
+        },
```

Comment:
I propose that we only send checksums of games that are relevant to this multiworld. That way special handling of the Archipelago pseudo-game can be removed, and we don't have to match datapackage_* to games to figure out what to request.

Reply:
Might not be a terrible idea actually, no need to send checksums for games not being played. I can definitely make that adjustment.